src/app/space/settings/areas/areas.component.ts

```diff
@@ -18,69 +43,251 @@ export class AreasComponent implements OnInit, OnDestroy {
+  selectedAreaId: string;
+  subscriptions: Subscription[] = [];
+  resultsCount: number = 0;
+  showTree: boolean = true;
+  treeListConfig: TreeListConfig;
+
+  constructor(private contexts: ContextService,
+              private areaService: AreaService) {
     this.contexts.current.subscribe(val => this.context = val);
   }
 
   ngOnInit() {
-    this.listConfig = {
+    this.actionConfig = {
+      moreActions: [{
+        id: 'addChildArea',
+        title: 'Add Child Area',
+        tooltip: 'Add Child Area'
+      }]
+    } as ActionConfig;
+
+    this.emptyStateConfig = {
+      actions: {
+        primaryActions: [{
+          id: 'addArea',
+          title: 'Add Area',
+          tooltip: 'Add Area'
+        }],
+        moreActions: []
+      } as ActionConfig,
+      title: 'Add Area',
+      info: 'Start by adding an area.'
+    } as EmptyStateConfig;
+
+    this.treeListConfig = {
       dblClick: false,
-      dragEnabled: false,
       emptyStateConfig: this.emptyStateConfig,
       multiSelect: false,
       selectItems: false,
       showCheckbox: false,
-      useExpandItems: false
-    } as ListConfig;
+      treeOptions: {
+        allowDrag: false,
+        isExpandedField: 'expanded'
+      }
+    } as TreeListConfig;
 
-    this.areaSubscription = this.areaService.getAllBySpaceId(this.context.space.id).subscribe(areas => {
+    this.subscriptions.push(this.areaService.getAllBySpaceId(this.context.space.id).subscribe(areas => {
       this.selectedAreaId = this.context.space.id;
       areas.forEach((area) => {
         if (area.attributes.parent_path == '/') {
           this.selectedAreaId = area.id;
           this.defaultArea = area.id;
         }
       });
-      this.areas = areas;
-    });
+      this.allAreas = areas as ExtArea[]; // store all areas for filter/sort
+      this.treeAreas = this.buildTree(this.allAreas); // transform flat array into tree list
+    }));
   }
 
   ngOnDestroy() {
-    this.areaSubscription.unsubscribe();
+    this.subscriptions.forEach(sub => {
+      sub.unsubscribe();
+    });
   }
 
-  openModal() {
+  openModal(): void {
     this.modal.show();
   }
 
-  addChildArea(id: string) {
+  addChildArea(id: string): void {
     this.selectedAreaId = this.defaultArea;
     if (id) {
       this.selectedAreaId = id;
     }
     this.openModal();
   }
 
-  itemPath(item: AreaAttributes) {
-    // remove slash from start of string
-    let parentPath = item.parent_path_resolved.slice(1, item.parent_path_resolved.length);
-    if (parentPath === '') {
-      return item.name;
+  addArea(area: ExtArea): void {
+    this.allAreas.push(area);
+    this.showTree = false;
+
+    // Hide the tree list to rerender existing toggles in expanded state
+    setTimeout(() => {
+      // Reapply sort and filter, if any
+      this.allAreas.sort((area1: Area, area2: Area) => this.compare(area1, area2));
+      this.applyFilters(this.appliedFilters);
+      this.showTree = true;
+    }, 10);
```

Comment:
What's the reason for a 10 ms timeout for these lines?

Reply:
When an area is added, the tree list is briefly hidden in order to re-render existing toggles in expanded state. This is related to angular-tree-component and Angular's change detection strategy.

Alternatively, if we don't show the toggles expanded, the timeout can be removed. Or, if we want to fetch new data and rebuild the tree, that may force a change detection? However, that is probably less performant than adding the area directly to the existing list.

Reply:
With the above fix to the `ngIf` usage, this function can be shortened to:

```
    this.allAreas.push(area);
    this.allAreas.sort((area1: Area, area2: Area) => this.compare(area1, area2));
    this.applyFilters(this.appliedFilters);
```

This will then function as you've described without having to force Angular CD to run via a timeout.

Reply:
Actually I think I misunderstood your intention.

The current behavior will expand 'all' unexpanded areas in the tree whenever an area is added anywhere. Is that the intention?

Reply:
Well, regardless, I don't think expanding all areas in the tree when I add an area is good behavior. It should at most expand the parent areas that my new area is added under to guarantee my new area is visible, or not expand at all. The former will require quote some changes to the code, the latter can be done with the suggested code above.

Reply:
Removed the functionality